From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Fri, 29 Dec 2023 12:23:23 +0100
Subject: [PATCH] Fiddle branding changes - License - Package into jar

Adds `LICENSE.txt` to built jar files.

This patch has a parallel patch with the same purpose and name in `fiddle-api`.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

This patch is based on the following patch:
"Gale branding changes"
By: Martijn Muijsers <martijnmuijsers@live.nl>
As part of: Gale (https://github.com/GaleMC/Gale)
Licensed (except for image files) under: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)

* Gale description *

Gale - https://galemc.org

diff --git a/build.gradle.kts b/build.gradle.kts
index f4e6c46c85850d16153f348e00534cef426c36b4..8acbf54dfb857a63b3228214af1d44fca55e3312 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -220,3 +220,22 @@ tasks.registerRunTask("runDev") {
     description = "Spin up a non-relocated Mojang-mapped test server"
     classpath(sourceSets.main.map { it.runtimeClasspath })
 }
+
+// Fiddle start - branding changes - license - package into jar
+tasks.register<Copy>("copyLicense") {
+    from(layout.projectDirectory.file("LICENSE.txt"))
+    into(layout.buildDirectory.dir("tmp/copiedlicense"))
+}
+
+tasks.processResources {
+    dependsOn("copyLicense")
+}
+
+sourceSets {
+    main {
+        resources {
+            srcDir(layout.buildDirectory.dir("tmp/copiedlicense"))
+        }
+    }
+}
+// Fiddle end - branding changes - license - package into jar
